packages/foam-vscode/src/core/utils/log.ts

```diff
@@ -1,3 +1,4 @@
+// eslint-disable-next-line @typescript-eslint/interface-name-prefix
```

Comment:
If we enable the rule, it's all about consistency, so it has two modes: use the `I` prefix always or never.

> could actually be replaced by a combo of interface Logger + abstract Class Logger

I think the class declaration will merge with the type defined by the interface declaration. It's not as if they lived in different namespaces (the namespace of interfaces and the namespace of classes). So AFAICS `interface Logger + abstract Class Logger` and `interface ILogger + abstract Class Logger` will have different implications for the type checker. For example:

```ts
interface C {
  a: number;
  
  b: string;
}

abstract class C {
  a: number;

  b: number;  // <-- Error: Subsequent property declarations must have the same type
}
```

or 

```ts
interface C {
  a: number;

  b: string;
}

abstract class C {
  a: number;

  b: string;

  c: number;
}

function f(c: C): C {
  return c;
}

f({ a: 0, b: '1' }); // <-- Error: c is missing
```

I've not found this strictly documented anywhere, but my understanding is that the class declaration will create both the runtime object and a compile time type that will merge with an existing type named the same, so changing the type created by the interface declaration as well.

Given that the rule is about consistency I see no point in disabling it on a case by case basis like I did. And refactoring will require coming up with different names for classes and interfaces. So let's disable this rule altogether?

Reply:
Sorry my example of `URI` is no longer valid, I was referring to it before the refactoring, where the `abstract class` only has static methods.

A good, current example is `Resource`

Reply:
Yes, I assumed you referred to the previous implementation. But my point is that we cannot just rename `ISomething` to `Something` without care, because any other `Something` class in the same namespace will suddenly clash with the interface. Both classes and interfaces will be adding to the same underlying type `Something`. And since I'm deliberately avoiding thoughtful refactorings here, I'm prone to disabling the rule.